File: 백준/Silver/11651. 좌표 정렬하기 2/좌표 정렬하기 2.js
let input = require('fs')
.readFileSync('/dev/stdin')
.toString().split('\n');

let number = Number(input[0]);
let arr = [];
for (let i = 1; i <= number; i++){
    let [x,y] = input[i].split(' ').map(Number);
    arr.push([x,y]);
}

arr.sort(
    function(a,b){
        if (a[1] != b[1]){
            return a[1] - b[1]
        }
        return a[0] - b[0]       
    }
)
let answer = "";
for (let item of arr){
    answer += item[0] + " " + item[1] + "\n";
}
console.log(answer);

File: 백준/Silver/11651. 좌표 정렬하기 2/좌표 정렬하기 2.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Text;

class Program
{
    static void Main()
    {
        using (StreamReader reader = new StreamReader(Console.OpenStandardInput()))
        using (StreamWriter writer = new StreamWriter(Console.OpenStandardOutput()))
        {
            int number = int.Parse(reader.ReadLine());
            List<int[]> arr = new List<int[]>(number);

            for (int i = 0; i < number; i++)
            {
                string[] input = reader.ReadLine().Split(' ');
                int x = int.Parse(input[0]);
                int y = int.Parse(input[1]);
                arr.Add(new int[] { x, y });
            }

            arr.Sort((a, b) => 
            {
                int compare = a[1].CompareTo(b[1]);
                if (compare != 0) return compare;
                return a[0].CompareTo(b[0]);
            });

            StringBuilder sb = new StringBuilder();
            foreach (var item in arr)
            {
                sb.AppendLine($"{item[0]} {item[1]}");
            }
            writer.Write(sb.ToString());
        }
    }
}
